.gitignore

```diff
@@ -8,3 +8,7 @@
 # Compiled binary
 /cmd/cloud_sql_proxy/cloud_sql_proxy
 /cloud_sql_proxy
+# v2 binary
+/cloudsql-proxy
```

Comment:
Yes, I agree. We'll need to update the build scripts to output `cloud_sql_proxy`. This is for local dev-flow convenience. If I run `go build` it will use the module name `cloudsql-proxy`. Granted, I could add the `-o cloud_sql_proxy` locally, but `go build` is much easier. And so I add this line here to avoid committing the dev artifact.

Reply:
FYI you can just make a local-only gitignore: https://stackoverflow.com/questions/5724455/can-i-make-a-user-specific-gitignore-file

Reply:
For peculiar dev workflows, I think a local-only gitignore is perfectly reasonable. Out of the box, a `go build` on the v2 branch will produce this exact binary. So I think we should ignore it globally. What if I added a comment to that effect, e.g., "Ignore the binary produced by building the module"?